### PR TITLE
Feat/9653 duration human readable arenav2 from main

### LINKED ISSUE #9653: [FEATURE] Display contest duration in human-readable format in ArenaV2
## Checklist

- [x] I have searched existing issues and confirmed this is not a duplicate (Required)

In `/arenav2/`, contest durations are not consistently displayed in a human-readable format. While longer contests (e.g., 30 days) are already shown correctly:

Image

shorter durations still need to be handled.

## Feature request

Implement human-readable formatting for contest durations across all time ranges in /arenav2/. For durations that are not exact or may vary slightly, approximate them to the closest meaningful unit to improve readability and user experience.

The implementation in PR #8412 can serve as a reference.

### PR DESCRIPTION
**Pull request by rahmanfaizur**
## Human-readable duration formatting in ArenaV2 contest cards

Adds a heuristic formatter for contest durations to improve readability:

- `< 1 hour` → exact minutes  
- `< 24 hours` → rounded hours  
- `< 30 days` → rounded days  
- `≥ 30 days` → months  

Updates ArenaV2 contest cards to use the new formatter.

Includes unit tests covering formatting rules and edge cases.

## Screenshots

Screenshot 1

Screenshot 2

Fixes #9653

### CODE REVIEW

File: frontend/www/js/omegaup/time.ts [hunk 1]
```diff
@@ -319,6 +319,65 @@ export function formatContestDuration(
...
+ * Heuristic:
+ * - < 1 hour: show minutes
+ * - < 24 hours: round to nearest hour (~X hours)
+ * - < 30 days: round to nearest day (~X days)
+ * - >= 30 days: convert to months (~X months)
+ *
+ * This intentionally differs from `formatContestDuration` which uses a
+ * mix of `formatDelta` (HH:MM:SS-style) and `date-fns/intervalToDuration`.
+ */
+export function formatContestDurationHumanReadable(
```
Comment (pabo99) [inline frontend/www/js/omegaup/time.ts:334]: How complicated would it be to add an extra validation?

Let me explain:

This works very well for displaying exact human-readable times, but what happens when contests have a duration like 10h:10m:10s? Your function shows “10 hours,” which is the rounded value. Would it be possible to add a `~` before the text when this happens?

If it’s going to make the calculations more complex, we can leave it as is. But if it’s feasible, could you also add it along with the corresponding tests?

File: frontend/www/js/omegaup/time.test.ts [hunk 2]
```diff
@@ -301,4 +301,59 @@ describe('time', () => {
       }
     });
   });
+
+  describe('formatContestDurationHumanReadable', () => {
+    it('Should format contest duration with human-readable units', () => {
```
Comment (pabo99) [inline frontend/www/js/omegaup/time.test.ts]: Would it be possible for you to use a parameterized test instead? With the suggestion I just left, more cases would be added, and this test would end up duplicating too many lines of code.

**Review by pabo99 — COMMENTED** [review]
This is great! I left you a couple of suggestions. Once you apply them, we can merge it.

File: frontend/www/js/omegaup/time.test.ts [hunk 3]
```diff
@@ -301,4 +301,32 @@ describe('time', () => {
...
     });
   });
+
+  describe('formatContestDurationHumanReadable', () => {
+    const minuteMs = 60 * 1000;
+    const hourMs = 60 * 60 * 1000;
+    const dayMs = 24 * 60 * 60 * 1000;
+    const start = new Date('2021-01-01 00:00:00+00:00');
+
+    it.each([
```
Comment (pabo99) [inline frontend/www/js/omegaup/time.test.ts:311]: Much better, thank you!
